pyinfra_docker/docker.py

```diff
@@ -60,25 +77,27 @@ def _yum_or_dnf_install(yum_or_dnf):
 
     yum_or_dnf.packages(
         name="Install Docker via yum",
-        packages=["docker-ce"],
+        packages=packages,
         extra_install_args=extra_install_args,
     )
 
 
 @deploy("Deploy Docker")
-def deploy_docker(config=None):
+def deploy_docker(config=None, version=None):
     """
     Install Docker on the target machine.
 
     Args:
         config: filename or dict of JSON data
     """
 
+    packages = get_pkgs_to_install(version)
```

Comment:
What about making `version` a host data, [maybe with default](https://docs.pyinfra.com/en/2.x/api/deploys.html#data-in-deploys), instead?

Reply:
I don't think we want to force the default to be an older package version for now, while it's having issues for us it might be very swiftly patched and anyone using this will be getting an older package 🙃

Reply:
The default could be `None` and the same logic then applies. It would remove the need to pass the version through the functions.
I'm not sure this is the best coding practice (prefer dependency injection over global variables) but it seems to me it would be  more in line with the spirit of `pyinfra`.

Reply:
Yeh I see your point!
[ca64934](https://github.com/Fizzadar/pyinfra-docker/pull/5/commits/ca6493442debcb943e72d5101d85fb2b15f607e8)

Reply:
> I'm not sure this is the best coding practice (prefer dependency injection over global variables) but it seems to me it would be more in line with the spirit of pyinfra.

heh, totally agree with this, the deploy data defaults thing is... not great. But absolutely right thing to do here now.